Bump `zip` from `1.1.4` to `2.2.2`

cakebaker opened the pull request:
This PR manually bumps `zip` from `1.1.4` to <del>`2.2.0`</del> <del>`2.2.1`</del>`2.2.2` because `renovate` seems to have a problem with updating this crate (https://github.com/uutils/coreutils/pull/6426).

sylvestre commented:
i would prefer this issue is fixed before upgrading:
https://github.com/zip-rs/zip2/pull/247

cakebaker commented:
Changes since last push:

* fixed merge conflict
* bumped `zip` from `2.2.0` to `2.2.1` (the issue mentioned in https://github.com/uutils/coreutils/pull/6873#issuecomment-2485065470 is still open)

cakebaker commented:
Changes since last push:
* bumped `zip` from `2.2.1` to `2.2.2`
